game/tournament.py

```diff
@@ -36,22 +44,53 @@ def get_results_table(self):
     def get_scores(self):
         return self._scores_dict
 
+    def timeout_handler(self, arg1, arg2):
+        raise TimeLimitReached
+
     def run(self):
         for match in self._matches:
-            match.play_full_match()
+            abort_current_match = False
+            penalized_player = None
+            signal.signal(signal.SIGALRM, self.timeout_handler)
+            for player in match.get_players():
+                signal.setitimer(signal.ITIMER_REAL,self._prepare_time_limit)
```

Comment:
I know, but since taking more time automatically disqualifies the strategy is not that bad. But as I said, I trust you if you want to go with this implementation